relentless/potential/potential.py

```diff
@@ -34,28 +131,123 @@ def evaluate(self, pair):
                 params[p] = v
             elif v is not None:
                 raise TypeError('Parameter type unrecognized')
-
             if v is None:
                 raise ValueError('Parameter {} is not set for ({},{}).'.format(p,pair[0],pair[1]))
 
         return params
 
     def save(self, filename):
-        all_params = {}
+        """Saves the data from a coefficient matrix to a file.
+
+        Serializes each parameter value in order to track the internal values
+        of the :py:class:`Variable` object and distinguish it from 'scalar' values.
+
+        Parameters
+        ----------
+        filename : `str`
+            The name of the file to save data in.
+
+        """
+        data = {}
         for key in self:
-            all_params[str(key)] = self.evaluate(key)
+            all_params = {}
+            for param in self[key]:
+                var = self[key][param]
+                if isinstance(var, Variable):
+                    all_params[param] = {'type':'variable', 'value':var.value,
+                                         'const':var.const, 'low':var.low, 'high':var.high}
+                elif isinstance(var, (float,int)):
+                    all_params[param] = {'type':'scalar', 'value':var}
+            data[str(key)] = all_params
         with open(filename, 'w') as f:
-            json.dump(all_params, f, sort_keys=True, indent=4)
+            json.dump(data, f, sort_keys=True, indent=4)
 
     def load(self, filename):
-        raise NotImplementedError('Load is not implemented')
+        """Loads the data from a file into the coefficient matrix.
+
+        Parameters
+        ----------
+        filename : `str`
+            The name of the file from which to load data.
+
+        Raises
+        ------
+        KeyError
+            If the pairs in the data are not exactly the pairs in self
+        KeyError
+            If the params in the data are not exactly the params in self
+        TypeError
+            If the parameter value type in the data and in self do not match
+
+        """
+        with open(filename, 'r') as f:
+            data = json.load(f)
```

Comment:
You end up coercing the keys twice, which is not ideal. Better would be to do something like
```py
data_ = json.load(f)
data = {eval(key): data_[key] for key in data_}
```
so that `data` is just keyed by tuples.